src/H5Fvfd_swmr.c

```diff
@@ -319,7 +350,20 @@ H5F_vfd_swmr_close_or_flush(H5F_t *f, hbool_t closing)
         if (H5F__vfd_swmr_update_end_of_tick_and_tick_num(shared, TRUE) < 0)
             HDONE_ERROR(H5E_FILE, H5E_CANTSET, FAIL, "unable to update end of tick");
     }
+#if 1 /*Kent  Save the end of close or flush info. to the log file, subject to comment out. */
+    H5F_POST_VFD_SWMR_LOG_ENTRY(f, 2, "VFD SWMR File close or flush ends");
+#endif
 done:
+
+    /* Kent: Stop the timer and close the VFD SWMR log file if it is turned on.
+     * Please REVIEW to ensure this is the right place to
+     * close the log file.
+     */
+    if (shared->vfd_swmr_log_on) {
```

Comment:
Do we close the log file if it is flushing not file closing?

Reply:
I don't know if this close the log file when the file is flushing.  As in the comments, I don't know if this is the right place to close the log file. If you know there is a place that I can close the log file, please let me know. I checked H5Fclose first but there is no H5F_t in that routine. So this is the best place I can find. If you are not sure, I suggest we keep here until you and John can find the right place to close the log file.

Reply:
Vailin, I think I find the solution. 
I can just add a check if this is for file closing:
Like this
 ```
if (shared->vfd_swmr_log_on && closing) {
       H5_timer_stop(&(shared->vfd_swmr_log_start_time));
        HDfclose(shared->vfd_swmr_log_file_ptr);
} 
```
Then the log file will only be closed when this function is handling the close of the file.

Reply:
ok